frontend/src/components/SignupForm.js

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { signup } from '../reducers/user';
+import { Wrapper, Form } from 'lib';
+
+const SignupForm = () => {
+  const dispatch = useDispatch();
+  const errorMessage = useSelector((store) => store.user.login.errorMessage);
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+
+  // This part can be improved with having one error object as a useState
```

Comment:
Could you explain what your idea is to improve this? We're interested to know how the error object would work.